### ISSUE TITLE
Implement warnings when comparing str and bytes

### ISSUE THREAD

**Issue by pfalcon** [micropython/micropython#2213]
In CPython3, this is enabled with -b switch:

~~~
>>> "b" == b"b"
__main__:1: BytesWarning: Comparison between bytes and string
~~~

This issue already is or will be one of the biggest head-scratchers, so I guess for baremetal ports, this should be default.

**Comment by dpgeorge** [comment]
Sounds good.

**Comment by pfalcon** [comment]
Implemented in 707cae749424104660cd24e7c1f1e16d67b99b0f and follow-up commits.
